### ISSUE TITLE
Migrate visualisations to Lens

### ISSUE THREAD

**Issue by ruflin** [elastic/integrations#4768]
Lens is the main editor that should be used for visualisations. Existing dashboards we own must be migrated to it. In https://github.com/elastic/integrations/issues/265 a previous discussion happened on Gaps to moving to Lens. The obvious ones are resolved but there might be more we uncover during the migration. This issue has the purpose to uncover these issues and track the migration to Lens. Any blockers during the migration must be mentioned in this issue. The current state of the migration from legacy visualisations to Lens for integrations can be found [here](https://visualization-usage-analytics.kb.us-central1.gcp.cloud.es.io:9243/app/dashboards#/view/8cf80710-6aa0-11ed-9c67-b36e35159418?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-90d%2Fd,to:now))) (private link).

On overview from the state of 2022-12-06 is shared on the bottom. For tracking, I will list the top 5:

* [x] #4994
* [ ] System @elastic/elastic-agent-data-plane
* [ ] Cef @elastic/security-external-integrations
* [x] #4995
* [x] #5015
* [ ] Network Traffic @elastic/security-external-integrations

For the migration, there is an "Open in Lens" button which should hopefully help with the migration. The overall goal is to migrate all visualisations but lets start with the top 5 as the initial priority to also uncover potential migration blockers.

![Screenshot 2022-12-06 at 16 01 40](https://user-images.githubusercontent.com/244900/205947438-34cffa05-6746-4201-b253-bac74c2b6ab3.png)

**Comment by ruflin** [comment]
@jsoriano Similar discussion applies here like for by reference. Can we add some hints during the build process or similar to warn devs?

**Comment by jsoriano** [comment]
> Similar discussion applies here like for by reference. Can we add some hints during the build process or similar to warn devs?

We need some better way to encourage developers to use more current standards. Printing warnings as we do now in some cases is easily ignored or overlooked. Producing errors in CI is usually too disrupting and difficult to introduce. We'd need to find some intermediary approach. Maybe https://github.com/elastic/package-spec/issues/342 can help, but is probably not enough.

We may need some mechanism to encourage warnings reduction in CI, maybe we can do something like failing if a PR affecting a package with warnings doesn't reduce the number of them.

**Comment by ruflin** [comment]
@jsoriano Agree. This might become a longer discussion, lets take it to a separate issue.

**Comment by andrewctate** [comment]
@gvnmagni is part-way through a refactor of the system metrics dash!

**Comment by kaiyan-sheng** [comment]
Which version of Kibana starts supporting "Open in Lens" button?

**Comment by andrewctate** [comment]
@kaiyan-sheng it was [first introduced](https://github.com/elastic/kibana/pull/114794) in 8.2. We have continued adding support for further configurations since then, especially in 8.5 and 8.6 when we finished all the cases we outlined for [phase 1](https://github.com/elastic/kibana/issues/138236). [This](https://github.com/elastic/kibana/issues/115034) is the overall meta-issue that links to everything.

So, a gradually increasing number of legacy visualizations are convertible as you advance from 8.2 forward to 8.6 where the vast majority can be (still subject to human review).

In 8.7, we'll be promoting "Open in Lens" on dashboards (you can see some screenshots on https://github.com/elastic/kibana/pull/146363) which will make it more obvious which visualizations can be automatically converted and easier to convert them.

**Comment by milanparmar-crest** [comment]
Team FYI, We're currently scoping this effort and we are planning to work on this issue.

**Comment by roshan-elastic** [comment]
We have a load of visualisations in the hosts view which would benefit from moving to Lens (I had an SRE interview today where they were expecting to be able to go into lens and add):

![image](https://user-images.githubusercontent.com/117740680/216034403-0ea62c75-4353-4237-ab3b-48d320e51c72.png)

![image](https://user-images.githubusercontent.com/117740680/216034409-d31d9e4a-ee9e-49b4-a42e-51d10f5e679c.png)


+1 on the effort (you can assume the Infra UI Team will prioritise this effort where it adds business value - I'm seeing this already)

**Comment by kush-elastic** [comment]
We have dashboards that contain Input Controls visualizations and are now deprecated. We are thinking of using new Controls ([introduced in 8.3](https://www.elastic.co/guide/en/kibana/8.3/add-controls.html)) as [part of this](https://github.com/elastic/integrations/issues/5221).
We found a few discrepancies between the Input Controls visualization and Controls panels as below.

Input Controls Visualization:
![image](https://user-images.githubusercontent.com/89848966/218722321-dd97a7eb-bd33-4ebc-932e-d75774b3612c.png)

Control:
![image](https://user-images.githubusercontent.com/89848966/218722896-6aca69e3-33a5-4255-8e67-c7a7542d680f.png)


Input Control visualization: We can make dependency filters that strictly depend on each other.
New Controls: Dependency filtering is possible but filters are not required to be strict.
In Input Control visualization, the default placeholder is `Select` while in new Controls it is `Any`. (I don't think it will affect users)

There shouldn't be any functional difference between these two panels.

**Comment by drewdaemon** [comment]
cc @ThomThomson ^^

**Comment by kush-elastic** [comment]
@ThomThomson sorry but I guess it was confusing, I just had a doubt that if we use new Controls it will users in any way or not.
If this new Controls were created in the manner of solving the disabling filters then i think it is right as well.
We will be using new Controls to replace older input controls.
Thank you 👍

**Comment by ThomThomson** [comment]
@kush-elastic, let us know if you run into any trouble, or have any feature requests or feedback for the new Controls!

**Comment by kush-elastic** [comment]
> @kush-elastic, let us know if you run into any trouble, or have any feature requests or feedback for the new Controls!

@ThomThomson 
Sure, Thanks for the quick response.
